docs/references/sdk/frontend-only.mdx

```diff
@@ -1,7 +1,152 @@
+
+> [!NOTE]
+> The code blocks below will be written in pseudo-code. If you're looking for real-world examples, have a look at these repositories: [`@clerk/clerk-react`](https://github.com/clerk/javascript/tree/main/packages/react), [`@clerk/astro`](https://github.com/clerk/javascript/tree/main/packages/astro)
+
+<Steps>
+  ### Create a Clerk instance
+
+  Create a Clerk instance that will only be invoked once (e.g. following the [singleton pattern](https://www.patterns.dev/vanilla/singleton-pattern/)). During its initialization you'll execute the following steps.
+
+  ```tsx {{ mark: [5], filename: 'create-clerk-instance.ts' }}
+  import { runOnce } from './utils'
+  // States accessible to other parts of your SDK and/or its users
+  import { $clerk, $state } from './stores'
+
+  export const createClerkInstance = runOnce(createClerkInstanceInternal)
+
+  async function createClerkInstanceInternal(options) {
+    let clerkJSInstance = window.Clerk
+  }
+  ```
+
+  ### Hotload ClerkJS
+
+  In order to make `Clerk` available on the `window` object, your SDK needs to load ClerkJS. Call `loadClerkJsScript()` from `@clerk/shared`.
+
+  ```tsx {{ mark: [1, [10, 17]], filename: 'create-clerk-instance.ts' }}
+  import { loadClerkJSScript } from '@clerk/shared/loadClerkJSScript'
+  import { runOnce } from './utils'
+  import { $clerk, $state } from './stores'
+
+  export const createClerkInstance = runOnce(createClerkInstanceInternal)
+
+  async function createClerkInstanceInternal(options) {
+    let clerkJSInstance = window.Clerk
+
+    if (!clerkJSInstance) {
+      await loadClerkJSScript(options)
+
+      if (!window.Clerk) {
+        throw new Error('Failed to download latest ClerkJS.')
+      }
+      clerkJSInstance = window.Clerk
+    }
+
+    if (!$clerk.get()) {
+      $clerk.set(clerkJSInstance)
+    }
+  }
+  ```
+
+  ### Call `window.Clerk.load()`
+
+  By calling [`window.Clerk.load()`](/docs/references/javascript/clerk/clerk#load) the Clerk class is initialized and your SDK now has access to all its functionality.
+
+  ```tsx {{ mark: [18], filename: 'create-clerk-instance.ts' }}
+  import { loadClerkJSScript } from '@clerk/shared/loadClerkJSScript'
+  import { runOnce } from './utils'
+  import { $clerk, $state } from './stores'
+
+  export const createClerkInstance = runOnce(createClerkInstanceInternal)
+
+  async function createClerkInstanceInternal(options) {
+    let clerkJSInstance = window.Clerk
+
+    if (!clerkJSInstance) {
+      // loadClerkJSScript() codepath
+    }
+
+    if (!$clerk.get()) {
+      $clerk.set(clerkJSInstance)
+    }
+
+    await clerkJSInstance.load(options)
+  }
+  ```
+
+  ### Add event listeners
+
+  Expose properties to the internal state (e.g. a `useState` in React) through adding [event listeners](/docs/references/javascript/clerk/clerk#add-listener).
+
+  ```tsx {{ mark: [[20, 27]], filename: 'create-clerk-instance.ts' }}
```

Comment:
In the future we may want to differentiate between `tsx` and `ts` in the UI, for example showing different icons for each. For this reason I think we should use `ts` here since it's explicitly a TS file, not TSX

```suggestion
  ```ts {{ mark: [[20, 27]], filename: 'create-clerk-instance.ts' }}
```